app/components/DocumentListItem.tsx

```diff
@@ -78,6 +83,23 @@ function DocumentListItem(
     !!document.title.toLowerCase().includes(highlight.toLowerCase());
   const canStar = !document.isArchived && !document.isTemplate;
 
+  const membershipType = document.membershipType;
```

Comment:
This logic leads to some off behavior, for example if you are viewing a collection which contains a starred document and then click on that document it now jumps the sidebar to the top to focus it in the starred section.

We can probably lose this and `membershipType` but retain the rest of the fix?

Reply:
I'm afraid we need the `membershipType` so that shared docs to an user and within a group can be focused correctly in the sidebar.

re: stars: it was intentional to prefer stars over the actual document - It looks nice from "Home" page but it's unpleasant from "Collection" page 😞 I can think of a couple solutions.
1. Lose the whole "stars" priority? This will mean the stars won't be focused when opened from the document list in home / collections.
2. Focus the stars only if it's opened from Home?

Reply:
The second option seems ideal from a UX point of view, you're right... how to prevent this becoming overly complex

Reply:
We can listen to `history` events and conditionally update the location state based on the context in which the navigation happens. I think this helps centralize the logic, although it could become a bit complex to account for various navigations, e.g shared -> starred, home -> starred, shared -> shared, etc.

Reply:
I think `DocumentListItem` can be aware of where it's being rendered, either through a prop or a provider/context – we only need to know home for now so lets not over complicate.

Reply:
That was my initial thought, but there are quite a few cases which we can cover to provide a better UX. e.g. Navigate within the starred context when there's a link from a starred document to another starred document - This happens inside editor event handler, let me see if it's possible to cover them with the current solution 🤔 .

Reply:
Just updated the PR by introducing a function `determineSidebarContext` that uses the document and the current sidebar context to determine the next one.
Please note that most of the changes in this PR is forwarding the location state in case of any in-doc history updates.